.github/workflows/coverall.yml

```diff
@@ -0,0 +1,20 @@
+on: ["push", "pull_request"]
```

Comment:
I didn't use this action so not sure what it is doing, but if it is suppose to put some updates on a PR, then it needs to be `pull_request_target` to make it work for forks

and about `push`, do we really need it? in theory PR should be fine?

Reply:
Well... I think I broke it somehow, cause I cannot get it to run the workflow anymore 😩

Reply:
Lol, I have to solve conflicts from master before workflow changes take effect 🤯

Reply:
@derberg I am not sure if we should change to `pull_request_target`. Currently, I am using this trigger, but PR's do not trigger the workflow on this PR 🤔 I am not sure if that means that the coverage bot wont add comments on the PRs as wanted...

Reply:
It wont as stated in the readme https://github.com/coverallsapp/github-action#coveralls-github-action
> When running on pull_request events, a comment will be added to the PR with details about how coverage will be affected if merged.

Reply:
> and about push, do we really need it? in theory PR should be fine?

We need push on the master branch for when PRs are merged. Otherwise, the repo has no stats to compare PR's with.

Reply:
> @derberg I am not sure if we should change to pull_request_target. Currently, I am using this trigger, but PR's do not trigger the workflow on this PR 🤔 I am not sure if that means that the coverage bot won't add comments on the PRs as wanted...

> It wont as stated in the readme https://github.com/coverallsapp/github-action#coveralls-github-action

we have to, we cannot trust their readme to much 😄 
the thing is, they add a comment, to do that they need a token, then need write-token, and on `pull_request` it is not available on a fork, fork has access to read-only token.

of course it means that you need to first merge the workflow to see if it works 🤷🏼 
This action is strange, you do not need to provide any Coveralls token to talk to it, just GH token. For me it means they are probably analyzing the scopes of the token, and if you do not have `write` you cannot call their API. I hope I explained my view good? Makes sense?

> We need push on the master branch for when PRs are merged. Otherwise, the repo has no stats to compare PR's with.

but you do not push on the master, not in this PR. Even `push` means that workflow will trigger after merge, and that is it, it will just run, not commit/push or anything like that, so nothing will be saved on `master`. Why did you remove https://github.com/asyncapi/generator-model-sdk/pull/120/commits/700f3773fad8a4c90284478cc655681dd29eee86#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L61 ? was it not needed?

Reply:
> For me it means they are probably analyzing the scopes of the token, and if you do not have write you cannot call their API. I hope I explained my view good? Makes sense?

Remember that they have a bot that comments, and with public repositories, it should not be a problem. We just need to ensure that coveralls get notified of coverage changes in PRs. And that can only be done using `pull_request`.

> but you do not push on the master, not in this PR. Even push means that workflow will trigger after merge, and that is it, it will just run, not commit/push or anything like that, so nothing will be saved on master

We need the coverall job to run when PRs are merged for coverall to add the updated coverage report.

> Why did you remove 700f377#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L61 ? was it not needed?

Because we should never manually use the library, we use it though github action.